### PR TITLE
Fix image paths so they work with path prefix

### PR DESCRIPTION
**Pull request by planger**
Referencing images with a URL starting with `/` will always default to the root. This however won't work if we host the page in a folder, as we do in the preview. As we use plain HTML tags in the markdown, probably the easiest fix is to use relative image links instead. The same applies to markdown links starting with `/`. They don't seem to take URL prefixes into account.

This change turns all root paths into relative paths.

### CODE REVIEW

**Comment by xai** [comment]
> @xai Can you please take a look at this PR and let me know what you think? If you find a better solution for fixing the links and images so they work in the previews, please let me know. Thank you!

Looks good to me, I am not aware of a better approach than relative links for the markdown pages.

However, it looks like we still have several pages that use hard-coded absolute links to the theia-ide.org domain, see `git grep '](https://theia-ide.org/'`. They won't be broken in the preview but link to the production site targets instead of ages from the preview. Should we change them also into relative links?

**Comment by planger** [comment]
Thanks @xai! Good idea! I've replaced them in a68be8f. Can you please verify whether all links still work?
I noticed there are a few broken links before this change, but I left them for now. We should fix them in a separate PR though.

File: src/docs/blueprint_documentation.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ However, as signing is highly dependent on your setup, see the [electron builder
...
+The Theia IDE is based on the Theia platform, which is a flexible and adaptable platform for build tools and IDEs. Therefore, you can adapt the feature set and general appearance of the Theia IDE to your custom requirements with almost no limits. The Theia platform provides two mechanism to add your custom extensions: VS Code extensions and Theia extensions. Please have a look at the [overview about Theia extension capabilities](extensions/) for details. When assembling a product such as the Theia IDE, you can freely decide, which VS Code extensions and Theia extensions are part of it and thereby influence the feature set of your custom product. The following two sections describe how to modify which [VS Code Extensions](#updating-bundled-vs-code-extensions) and which [Theia extensions](#customizing-theia-extensions) are part of your product. Please also note that you can allow users of a Theia-based tool to [install VS Code extensions at runtime](user_install_vscode_extensions/).
```
Comment (xai) [inline src/docs/blueprint_documentation.md:60]:
```suggestion
The Theia IDE is based on the Theia platform, which is a flexible and adaptable platform for build tools and IDEs. Therefore, you can adapt the feature set and general appearance of the Theia IDE to your custom requirements with almost no limits. The Theia platform provides two mechanism to add your custom extensions: VS Code extensions and Theia extensions. Please have a look at the [overview about Theia extension capabilities](../extensions/) for details. When assembling a product such as the Theia IDE, you can freely decide, which VS Code extensions and Theia extensions are part of it and thereby influence the feature set of your custom product. The following two sections describe how to modify which [VS Code Extensions](#updating-bundled-vs-code-extensions) and which [Theia extensions](#customizing-theia-extensions) are part of your product. Please also note that you can allow users of a Theia-based tool to [install VS Code extensions at runtime](../user_install_vscode_extensions/).
```

File: src/docs/enhanced_tab_bar_preview.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ To also ensure the text is not going over the boxes boundaries, the property `wo
 
 After those rules are applied the preview will look as shown below:
 
-<img src="/enhanced-preview-custom.png" alt="A screenshot of the customized enhanced preview in Theia" style="max-width: 525px">
+<img src="../enhanced-preview-custom.png" alt="A screenshot of the customized enhanced preview in Theia" style="max-width: 525px">
```
Comment (xai) [inline src/docs/enhanced_tab_bar_preview.md:93]:
```suggestion
<img src="../../enhanced-preview-custom.png" alt="A screenshot of the customized enhanced preview in Theia" style="max-width: 525px">
```

File: src/docs/json_rpc.md [hunk 3]
```diff
@@ -90,7 +90,7 @@ an when this contribution is initialized it creates a websocket channel for all
 To save resources the hood all `MessagingContributions` are routed over one
 websocket connection (multiplexing).
 
-To dig more into ContributionProvider see this [section](Services_and_Contributions#contribution-providers).
+To dig more into ContributionProvider see this [section](../Services_and_Contributions#contribution-providers).
```
Comment (xai) [inline src/docs/json_rpc.md:93]:
```suggestion
To dig more into ContributionProvider see this [section](../services_and_contributions#contribution-providers).
```

File: src/docs/json_rpc.md [hunk 4]
```diff
@@ -221,7 +221,7 @@ Here we're creating a watcher, this is used to get notified about events
 from the backend by using the `taskWatcher`'s client
 (`taskWatcher.getTaskClient()`)
 
-See more information about how events work in theia [here](/docs/Events#events).
+See more information about how events work in theia [here](../Events#events).
```
Comment (xai) [inline src/docs/json_rpc.md:224]:
```suggestion
See more information about how events work in theia [here](../events#events).
```

File: src/docs/user_getting_started.md [hunk 5]
```diff
@@ -5,18 +5,18 @@ title: Using Theia As An End User
...
 
 The Theia IDE features and usability concept is heavily influences by VS Code. Therefore, for many use cases, we refer to the VS Code documentation. We still provide some selected documentation about use cases which are differing or not supported in VS Code:
 
-- [Install VS Code extensions](/docs/user_install_vscode_extensions/)
-- [Using the dynamic Toolbar](/docs/user_toolbar/)
+- [Install VS Code extensions](../user_install_vscode_extensions/)
+- [Using the dynamic Toolbar](../user_toolbar/)
 
-Please note that you can also [use the Theia IDE as a basis to create your own custom product](/docs/blueprint_documentation/).
+Please note that you can also [use the Theia IDE as a basis to create your own custom product](blueprint_documentation/).
```
Comment (xai) [inline src/docs/user_getting_started.md:22]:
```suggestion
Please note that you can also [use the Theia IDE as a basis to create your own custom product](../blueprint_documentation/).
```

**Review by xai — CHANGES_REQUESTED** [review]
Looks pretty good, just very few broken links left.

File: src/docs/getting_started.md [hunk 6]
```diff
@@ -5,21 +5,21 @@ title: Getting Started
...
 
 ## Build a tool or IDE based on Eclipse Theia
 
-The first step is to define a custom product based on Theia. Please check out the guide on [how to build your own IDE or tool based on Theia](https://theia-ide.org/docs/composing_applications/). Subsequently, you might want to extend this product with your own feature for which you can use the [available extension mechanisms of Theia](https://theia-ide.org/docs/extensions/).
+The first step is to define a custom product based on Theia. Please check out the guide on [how to build your own IDE or tool based on Theia](../composing_applications/). Subsequently, you might want to extend this product with your own feature for which you can use the [available extension mechanisms of Theia](../extensions/).
 
 ## Frequently Asked Questions
 
-If you are unsure what Theia is, how it compares to other technologies or how to migrate Eclipse-based tools to it, please have a look at [resource section](https://theia-ide.org/resources/).
+If you are unsure what Theia is, how it compares to other technologies or how to migrate Eclipse-based tools to it, please have a look at [resource section](../resources/).
```
Comment (xai) [inline src/docs/getting_started.md:20]:
```suggestion
If you are unsure what Theia is, how it compares to other technologies or how to migrate Eclipse-based tools to it, please have a look at [resource section](../../resources/).
```

File: src/docs/getting_started.md [hunk 7]
```diff
@@ -5,21 +5,21 @@ title: Getting Started
...
 
 
 ## Need Help?
 
-Theia is an open community, and we are glad to help you. You can ask questions using our [public community forum](https://github.com/eclipse-theia/theia/discussions), you can [report bugs and feature requests](https://github.com/eclipse-theia/theia/issues/new/choose), and you can get [professional support, consulting and implementation services](https://theia-ide.org/support/) for building products based on Theia.
+Theia is an open community, and we are glad to help you. You can ask questions using our [public community forum](https://github.com/eclipse-theia/theia/discussions), you can [report bugs and feature requests](https://github.com/eclipse-theia/theia/issues/new/choose), and you can get [professional support, consulting and implementation services](../support/) for building products based on Theia.
```
Comment (xai) [inline src/docs/getting_started.md:25]:
```suggestion
Theia is an open community, and we are glad to help you. You can ask questions using our [public community forum](https://github.com/eclipse-theia/theia/discussions), you can [report bugs and feature requests](https://github.com/eclipse-theia/theia/issues/new/choose), and you can get [professional support, consulting and implementation services](../../support/) for building products based on Theia.
```

**Review by xai — CHANGES_REQUESTED** [review]
Caught another two, sorry.

**Comment by planger** [comment]
Thank you, I think I addressed the additional ones you found. Can you please do one more check? It clearly seems four eyes are better, even though I'm using a link checker :-)

**Review by xai — APPROVED** [review]
LGTM! All broken links have been fixed.

**Review by JonasHelming — APPROVED** [review]
LGTM!

**Pull request merged**